webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/TableRow.js

```diff
@@ -0,0 +1,287 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Tr, Td, TableText } from '@patternfly/react-table';
+import {
+  TimesIcon,
+  CheckIcon,
+  PencilAltIcon,
+  FlagIcon,
+} from '@patternfly/react-icons';
+import {
+  SelectVariant,
+  Select,
+  SelectOption,
+  Button,
+  TextInput,
+  TextArea,
+  Checkbox,
+  FormGroup,
+  Form,
+  Tooltip,
+} from '@patternfly/react-core';
+import { APIActions } from '../../../../redux/API';
+import { sprintf, translate as __ } from '../../../../common/I18n';
+import { HOST_PARAM, columnNames, typeOptions } from './ParametersConstants';
+import { updateHost } from '../../ActionsBar/actions';
+import { RowActions } from './RowActions';
+
+export const ParametersTableRow = ({
+  param,
+  rowIndex,
+  editingRow,
+  setEditingRow,
+  hostId,
+  editHostsPermission,
+  isNew,
+}) => {
+  const [name, setName] = useState(param.name);
+  const [type, setType] = useState(param.parameter_type);
+  const [value, setValue] = useState(param.value);
+  const [selectIsOpen, setSelectIsOpen] = useState(false);
+  const [selectValueIsOpen, setSelectValueIsOpen] = useState(false);
+  const [isHide, setIsHide] = useState(param['hidden_value?']);
+
+  const dispatch = useDispatch();
+  const getValue = () => {
```

Comment:
If you pass in `param` and then place this function outside the component, it will be better because the function won't need to be recreated on every render.

Reply:
Thanks, moved it out